internal/services/managedhsm/parse/managed_hsm_key_id.go

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) HashiCorp, Inc.
```

Comment:
to simplify this can we move all the ID parser/id changes into a seperate PR so it can be more easily reviewed & unblock other PRs & work waiting on it?

Reply:
@wuxu92 can we also include the `ManagedHSMNestedKey` in the other PR

Reply:
@wuxu92 

1. `KeyId` should become `ManagedHSMNestedKeyWithVersionId`
2. `KeyVersionlessId` should become `ManagedHSMNestedKeyId`
3. `ManagedHSMKeyID` can be removed

-> Note: whilst this differs from the Key Vault Nested Item parsers - those will change to this approach in the future - since we *should not* be using these generic Resource ID parser methods going forward.

However as @katbyte has mentioned, can we split this out to a separate PR, since these Resource ID parsers need to be used elsewhere?